docs/validation.rst

```diff
@@ -0,0 +1,173 @@
+Validation
+----------
+
+By default, no validation rules are loaded or attached to the table. You must
+explicitly load the validation provider(s) and attach each rule if needed.
+
+Installation
+^^^^^^^^^^^^
+
+This plugin allows you to only load the validation rules that cover you needs.
+At this point there are 3 validation providers:
+    
+    - UploadValidation (validation rules useful for any upload)
+    - ImageValidation (validation rules specifically for images)
+    - DefaultValidation (loads all of the above)
+    
+Since by default, no validation rules are loaded, you should start with that:
+
+.. code:: php
+
+    <?php
+    
+        $validator->provider('upload', \Josegonzalez\Upload\Validation\UploadValidation::class);
+        // OR
+        $validator->provider('upload', \Josegonzalez\Upload\Validation\ImageValidation::class);
+        // OR
+        $validator->provider('upload', \Josegonzalez\Upload\Validation\DefaultValidation::class);
+        
+    ?>
+    
+Afterwards, you can use its rules like:
+
+.. code:: php
+
+    <?php
```

Comment:
This is how its done in 2.x: https://github.com/josegonzalez/cakephp-upload/blob/2.x/docs/validation.rst